exec/java-exec/src/test/java/org/apache/drill/exec/client/DrillClientSystemTest.java

```diff
@@ -73,4 +77,90 @@ public void testSubmitPlanTwoNodes() throws Exception {
     }
     client.close();
   }
+
+  @Test
+  public void testPopulateEndpointsList() throws Exception{
+
+    ArrayList<DrillbitEndpoint> endpointsList = new ArrayList<>();
+    String drillBitConnection;
+    DrillClient client = new DrillClient();
+    DrillbitEndpoint endpoint;
+    Iterator<DrillbitEndpoint> endpointIterator;
+
+
+    // Test with single drillbit ip
+    drillBitConnection = "10.10.100.161";
+    client.populateEndpointsList(endpointsList, drillBitConnection);
+    endpoint = endpointsList.iterator().next();
+    assert(endpointsList.size() == 1);
+    assert(endpoint.getAddress().equalsIgnoreCase(drillBitConnection));
+    assert(endpoint.getUserPort() == client.getConfig().getInt(ExecConstants.INITIAL_USER_PORT));
+
+    // Test with single drillbit ip:port
+    endpointsList.clear();
```

Comment:
Having to remember `endpointsList.clear( )` every time is error-prone. Better:

  {
      String drillBitConnection = "10.10.100.161";
      List<DrillEndpoint> endpointList = parseEndpoints( drillbitConnection );
      ... // Your tests here
    }

The above creates a new block, and new case-specific variables, of each specific test. Less chance of one test "bleeding through" into the next one.